src/main/kotlin/lotto/core/WinningRank.kt

```diff
@@ -16,7 +16,17 @@ enum class WinningRank(val winningCount: Pair<Int, Boolean>, val winningAmount:
             winningCount: Int,
             matchBonus: Boolean,
         ): WinningRank {
-            return RANK_BY_WINNING_COUNT[winningCount to matchBonus] ?: NOTHING
+            var ranks = entries.filter { it.winningCount == winningCount }
+            if (ranks.isEmpty()) {
+                return NOTHING
+            }
+
+            if (ranks.size == 1) {
+                return ranks[0]
+            }
+
+            ranks = ranks.filter { it.matchBonus == matchBonus }
+            return if (ranks.size > 0) ranks[0] else NOTHING
```

Comment:
아래와 같이 변경해보면 어떨까요? 🤔

```suggestion
            val ranks = entries.filter { it.winningCount == winningCount }
            return when(ranks.size) {
                0 -> NOTHING
                1 -> ranks.first()
                else -> ranks.firstOrNull { it.matchBonus == matchBonus } ?: NOTHING
            }
```

Reply:
간결해지는 것 확인했습니다.
반영하겠습니다.